src/main/java/com/synopsys/integration/alert/web/api/job/JobConfigActions.java

```diff
@@ -56,150 +62,183 @@
+    protected Optional<JobFieldModel> findJobFieldModel(UUID id) {
+        try {
+            Optional<ConfigurationJobModel> jobConfiguration = configurationAccessor.getJobById(id);
+            if (jobConfiguration.isPresent()) {
+                JobFieldModel jobFieldModel = readJobConfiguration(jobConfiguration.get());
+                return Optional.of(jobFieldModel);
+            }
+        } catch (AlertException ex) {
+            logger.error(String.format("Error finding job configuration id: %s", id), ex);
         }
         return Optional.empty();
     }
 
-    public List<JobFieldModel> getAllJobs() throws AlertException {
-        List<ConfigurationJobModel> allJobs = configurationAccessor.getAllJobs();
-        List<JobFieldModel> jobFieldModels = new LinkedList<>();
-        for (ConfigurationJobModel configurationJobModel : allJobs) {
-            JobFieldModel jobFieldModel = readJobConfiguration(configurationJobModel);
-            jobFieldModels.add(jobFieldModel);
+    @Override
+    protected ActionResponse<List<JobFieldModel>> readAllAfterChecks() {
+        try {
+            List<ConfigurationJobModel> allJobs = configurationAccessor.getAllJobs();
+            List<JobFieldModel> jobFieldModels = new LinkedList<>();
+            for (ConfigurationJobModel configurationJobModel : allJobs) {
+                JobFieldModel jobFieldModel = readJobConfiguration(configurationJobModel);
+                jobFieldModels.add(jobFieldModel);
+            }
+            return new ActionResponse<>(HttpStatus.OK, jobFieldModels);
+        } catch (AlertException ex) {
+            logger.error("Error reading all jobs", ex);
+            return new ActionResponse<>(HttpStatus.INTERNAL_SERVER_ERROR, ex.getMessage());
         }
-        return jobFieldModels;
     }
 
-    public void deleteJobById(UUID id) throws AlertException {
-        Optional<ConfigurationJobModel> jobs = configurationAccessor.getJobById(id);
-        if (jobs.isPresent()) {
-            LinkedList<FieldModel> processedFieldModels = new LinkedList<>();
-            ConfigurationJobModel configurationJobModel = jobs.get();
-            for (ConfigurationModel configurationModel : configurationJobModel.getCopyOfConfigurations()) {
-                FieldModel convertedFieldModel = modelConverter.convertToFieldModel(configurationModel);
-                FieldModel fieldModel = fieldModelProcessor.performBeforeDeleteAction(convertedFieldModel);
-                processedFieldModels.add(fieldModel);
-            }
-            configurationAccessor.deleteJob(configurationJobModel.getJobId());
-            for (FieldModel fieldModel : processedFieldModels) {
-                fieldModelProcessor.performAfterDeleteAction(fieldModel);
+    @Override
+    protected ActionResponse<JobFieldModel> deleteAfterChecks(UUID id) {
+        try {
+            Optional<ConfigurationJobModel> jobs = configurationAccessor.getJobById(id);
+            if (jobs.isPresent()) {
```

Comment:
If this is missing, I believe we should return a 404 to conform to the RFC standard.

Reply:
We already check if the object is present and will return a 404 not found in the delete method of AbstractJobResourceAction.  The check for if present is because the getJobById returns an optional.